docs/src/components/useCustomRouter.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+
+import { useRouter } from 'next/router';
+
+/**
+ *
+ * @returns same router format as Next.js with validated parameters.
+ * TODO: Next.js has a RFC to custom routes and validate params https://github.com/vercel/next.js/discussions/9081#discussioncomment-21895
+ * Once that feature is release, we may consider to migrate to their official ones instead.
+ */
+
+export const useCustomRouter = () => {
+  const router = useRouter();
+  const {
+    query: { platform = 'react' },
+    push,
+  } = router;
+  const platforms = ['react', 'angular', 'vue', 'flutter'];
```

Comment:
Yes, I've been thinking about it and also made the same comment in this PR https://github.com/aws-amplify/amplify-ui/pull/1892

To prevent introducing too many changes in the same PR, I didn't do it here